forwarder.go

```diff
@@ -226,28 +227,76 @@ func buildExternalURL(newTalariaName, domain string) string {
+func base64Decode(encodedStr string) ([]byte, error) {
+	decodedData, err := base64.StdEncoding.DecodeString(encodedStr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode base64 string: %v", err)
+	}
+	return decodedData, nil
+}
+
+// extractWebPAConveyHeader extracts the X-WebPA-Convey header value.
+func extractWebPAConveyHeader(req *http.Request) (string, error) {
+	xWebPACode := req.Header.Get("X-WebPA-Convey")
+	if xWebPACode == "" {
+		return "", fmt.Errorf("X-WebPA-Convey header not found")
+	}
+	return xWebPACode, nil
+}
+
+// updateResourceIpAddressAndCertificateInfo updates resource information.
+func updateResourceIpAddressAndCertificateInfo(req *http.Request, client *http.Client, resourceURL *url.URL) error {
+	certificateProviderRaw := req.Header.Get("X-Certificate-Provider")
 	certificateProviderType := "DTSECURITY"
 	if strings.Contains(certificateProviderRaw, "C2") {
 		certificateProviderType = "IRDETO"
 	}
 
+	// Extract X-WebPA-Convey header
+	xWebPACode, err := extractWebPAConveyHeader(req)
+	if err != nil {
+		return err
+	}
+
+	// Decode and unmarshal the extracted X-WebPA-Convey header
+	decodedData, err := base64Decode(xWebPACode)
+	if err != nil {
+		return err
+	}
+
+	var webPAData WebPAHeaderData
+	if err := json.Unmarshal(decodedData, &webPAData); err != nil {
+		return fmt.Errorf("failed to unmarshal decoded X-WebPA-Convey data: %v", err)
+	}
+	bootTime := time.Unix(webPAData.BootTime, 0)
+	// Prepare the request body
 	requestBody := UpdateResourceRequest{
-		IpAddress:               req.Header.Get(realIpHeader),
-		CertificateProviderType: certificateProviderType,
-		CertificateExpiryDate:   req.Header.Get(expiryDateHeader),
+		IpAddress:                req.Header.Get("X-Real-IP"),
+		CertificateProviderType:  certificateProviderType,
+		CertificateExpiryDate:    req.Header.Get(expiryDateHeader),
+		HwLastRebootReason:       webPAData.HwLastRebootReason,
+		WebpaInterfaceUsed:       webPAData.WebpaInterfaceUsed,
+		WebpaLastReconnectReason: webPAData.WebpaLastReconnectReason,
+		WebpaProtocol:            webPAData.WebpaProtocol,
+		LastBootTime:             bootTime,
+		FirmwareVersion:          webPAData.FwName,
 	}
 
-	log.Ctx(req.Context()).Info().Msgf("Certificate Provider type : [%s], Certificate expiry type : [%s]", requestBody.CertificateProviderType, requestBody.CertificateExpiryDate)
-	cpeIdentifier := strings.ToLower(req.Header.Get(deviceCNHeader))
+	// Log the details
+	log.Ctx(req.Context()).Info().Msgf("Certificate Provider type: [%s], Certificate expiry date: [%s], HW Last Reboot Reason: [%s], Webpa Interface Used: [%s], Webpa Last Reconnect Reason: [%s], Webpa Protocol: [%s]",
+		requestBody.CertificateProviderType, requestBody.CertificateExpiryDate,
+		requestBody.HwLastRebootReason, requestBody.WebpaInterfaceUsed,
+		requestBody.WebpaLastReconnectReason,
+		requestBody.WebpaProtocol)
+
+	cpeIdentifier := strings.ToLower(req.Header.Get("X-Device-CN"))
```

Comment:
What is the reason to remove constant from here and instead hardcode X-Device-CN?